### ISSUE TITLE
:add and :config of app_pool does not change the pipeline_mode

### ISSUE THREAD

**Issue by keyrage** [opscode-cookbooks/iis#164]
IIS cookbook 4.1 creates the pool but does not set the pool to "Classic" mode. The `:config` action is reporting in the chef debug logs that there is nothing to change. Multiple runs return the same result for `:config`.

- IIS 8.5
- Chef Client 12.0.1

- Attributes
```ruby
default['test_iis']['app_pool'] = 'TestAppPool'
default['test_iis']['app_pool_dotnetversion'] = "2.0"
default['test_iis']['pipelinemode'] = :Classic
default['test_iis']['thirty_two_bit'] = true
```

- Recipe
```ruby
iis_pool "#{node[''test_iis']['app_pool']}" do
  action [:add,:config]
  pipeline_mode node['test_iis']['pipelinemode']
  thirty_two_bit node[''test_iis']['thirty_two_bit']
  runtime_version node[''test_iis']['app_pool_dotnetversion']
end
```

**Comment by EasyAsABC123** [comment]
@keyrage Oh no! looking through the code right now, i'll be write back.

Just as a quick verification can you do the following in your recipe?
```ruby
log "app_pool = #{node[''test_iis']['app_pool']}"
log "pipelinemode = #{node['test_iis']['pipelinemode']}"
log "thirty_two_bit = #{node[''test_iis']['thirty_two_bit']}"
log "app_pool_dotnetversion = #{node[''test_iis']['app_pool_dotnetversion']}"
```

**Comment by EasyAsABC123** [comment]
So, here's the skinny:
`:add` is the only time that pipeline_mode is set...which isn't right, fixing it right now

**Comment by keyrage** [comment]
OK, I'm actually running into some other similar issues as well with iis_site resource where the application_pool under :config will not be detected as needing to be changed as well.

**Comment by EasyAsABC123** [comment]
please test pull request #165, hopefully it resolves the pool issue

**Comment by EasyAsABC123** [comment]
@keyrage you are correct about the application_pool, `:config` ignores it.

**Comment by EasyAsABC123** [comment]
please also test #167 as it should fix the application_pool bug :)

**Comment by keyrage** [comment]
That's crashing the chef run.

"NameError: No resource, method, or local variable named `returns' for `Chef::Provider::IisPool"

**Comment by keyrage** [comment]
config.rb provider code is buggy. Line 30 "shell_out!(cmd, returns: new_resource.returns)" should be "shell_out!(cmd, :returns => new_resource.returns)"

**Comment by keyrage** [comment]
:add now works and sets the pipeline_mode correctly, :config still doesn't detect or make the change.

**Comment by keyrage** [comment]
Getting this error:

ERROR: Converge failed with error message iis_pool[VenusAppPool] (poal_venuswebsite::default line 12) had an error: Mixlib::ShellOut::ShellCommandFailed: Expected process to exit with [0], but received '13'
---- Begin output of C:\Windows\System32\inetsrv\appcmd.exe set config /section:applicationPools "/[name='VenusAppPool'].autoStart:true" "/[name='VenusAppPool'].startMode:OnDemand" "/[name='VenusAppPool'].managedRuntimeVersion:v2.0" "/[name='VenusAppPool'].enable32BitAppOnWin64:true" "/[name='VenusAppPool'].queueLength:1000" "/[name='VenusAppPool'].processModel.maxProcesses:" "/[name='VenusAppPool'].processModel.loadUserProfile:false" "/[name='VenusAppPool'].processModel.logonType:LogonBatch" "/[name='VenusAppPool'].processModel.manualGroupMembership:false" "/[name='VenusAppPool'].processModel.idleTimeout:00:20:00" "/[name='VenusAppPool'].processModel.shutdownTimeLimit:00:01:30" "/[name='VenusAppPool'].processModel.startupTimeLimit:00:01:30" "/[name='VenusAppPool'].processModel.pingingEnabled:true" "/[name='VenusAppPool'].processModel.pingInterval:00:00:30" "/[name='VenusAppPool'].processModel.pingResponseTime:00:01:30" "/[name='VenusAppPool'].recycling.periodicRestart.time:" "/+[name='VenusAppPool'].recycling.periodicRestart.schedule.[value='']" "/[name='VenusAppPool'].recycling.logEventOnRecycle:PrivateMemory,Memory,Schedule,Requests,Time,ConfigChange,OnDemand,IsapiUnhealthy" "/[name='VenusAppPool'].recycling.periodicRestart.privateMemory:" "/[name='VenusAppPool'].recycling.disallowRotationOnConfigChange:false" "/[name='VenusAppPool'].recycling.disallowOverlappingRotation:false" "/[name='VenusAppPool'].failure.loadBalancerCapabilities:HttpLevel" "/[name='VenusAppPool'].failure.orphanWorkerProcess:false" "/[name='VenusAppPool'].failure.orphanActionExe:" "/[name='VenusAppPool'].failure.orphanActionParams:" "/[name='VenusAppPool'].failure.rapidFailProtection:true" "/[name='VenusAppPool'].failure.rapidFailProtectionInterval:00:05:00" "/[name='VenusAppPool'].failure.rapidFailProtectionMaxCrashes:5" "/[name='VenusAppPool'].failure.autoShutdownExe:" "/[name='VenusAppPool'].failure.autoShutdownParams:" "/[name='VenusAppPool'].cpu.action:NoAction" "/[name='VenusAppPool'].cpu.limit:0" "/[name='VenusAppPool'].cpu.resetInterval:00:05:00" "/[name='VenusAppPool'].cpu.smpAffinitized:false" "/[name='VenusAppPool'].cpu.smpProcessorAffinityMask:4294967295" "/[name='VenusAppPool'].cpu.smpProcessorAffinityMask2:4294967295" ----
STDOUT: ERROR ( message:Can not set attribute "time" to value "".. Reason: Timespan must be of form [-][d.]hh:mm:ss[.ff]

**Comment by keyrage** [comment]
"/+[name='VenusAppPool'].recycling.periodicRestart.schedule.[value='']" seems to be the offender.

**Comment by ksubrama** [comment]
@keyrage   shouldn't returns: new_resource.returns be equivalent to :returns => new_resource.returns ?  I believe that is the Ruby 1.9 syntax for hashes.  I merged your syntax fix (the :returns part was definitely broken) to unblock you, but I've also merged in @EasyAsABC123 's changes.  I think merging his change auto-closed this issue.  I'm manually reopening it while I investigate the schedule.[value=''] issue.

**Comment by ksubrama** [comment]
Ok - I'm now caught up on the other threads where this returns: thingy got discussed.  I'm going to try out some of the changes locally.  If the syntax changes seem to be causing more issues, I might revert our rubocop related changes for now.

**Comment by EasyAsABC123** [comment]
closing this as another issue was opened for schedule.[value='']
